vault/client.go

```diff
@@ -39,8 +40,23 @@ func CreateClient() (*api.Client, error) {
 		return nil, fmt.Errorf("while initializing Vault client: %w", err)
 	}
 
-	// always prefer VAULT_TOKEN environment variable
-	if os.Getenv("VAULT_TOKEN") == "" {
+	token := client.Token()
+
+	if token == "" {
+		helper, err := cliconfig.DefaultTokenHelper()
+		if err != nil {
+			return nil, fmt.Errorf("failed to get token helper: %w", err)
+		}
+		token, err = helper.Get()
```

Comment:
The library hides what happens here in an interface and I didn't find where the actual code is. What happens now if I have no config and not token at all?

Reply:
`cliconfig.DefaultTokenHelper` creates either an internal or an external token helper [here](https://github.com/hashicorp/vault/blob/main/api/cliconfig/util.go#L12). The internal token helper is the variant a) (i.e. reading/writing the token to the ~/.vault-token file), the external token helper is the variant b) (i.e. the execution of the script configured in the config as `token_helper`)